flower/events.py

```diff
@@ -103,6 +83,84 @@ def event(self, event):
         if cls:
             cls.send_message(event)
 
+    def remove_metrics_for_offline_workers(self):
+        if options.purge_offline_workers is not None:
+            offline_workers = self.get_offline_workers(workers=self.get_workers())
+            if not offline_workers:
+                return
+
+            self.metrics.remove_metrics_for_offline_workers(
+                offline_workers=offline_workers
+            )
+
+    def get_online_workers(self) -> Dict[str, Any]:
```

Comment:
You probably notice I started introducing typing into the project. 
The benefits are:
- any modern IDE gives you nice autocompletion
- easier reading/understanding inputs and outputs especially between multiple methods/objects